product-service/functions/getProductsById.ts

```diff
@@ -0,0 +1,30 @@
+import { APIGatewayProxyHandler } from 'aws-lambda';
+import productsList from '../mocks/productList';
+
+const fetchProductById = async (id) => {
+    const products = await Promise.resolve(productsList);
+    const product = products.find(product => product.id === id);
+
+  return product;
+}
+
+const getProductById: APIGatewayProxyHandler = async event => {
+    const product = await fetchProductById(event.queryStringParameters.id);
```

Comment:
You can use `pathParameters.id` if you get the `id` from the path parameters. Usually, it is passed not like a query but like path param.